services/posting.js

```diff
@@ -28,8 +28,8 @@ const createPost = async (req, res) => {
 
 const getPosts = async (req, res) => {
     await authenticateUser(req, res)
-    const { userId } = req.user
-    const post = await Posting.find({ userId })
+    const { username } = req.body
+    const post = await Posting.find({ username })
```

Comment:
this is wrong we want to use the userId
Its probably this syntax we need
```
const { userId } = req.user
    const post = await Posting.find({ createdBy: userId })
    ```
    Try that